packages/dd-trace/src/appsec/index.js

```diff
@@ -117,34 +121,49 @@ function incomingHttpEndTranslator ({ req, res }) {
     payload[addresses.HTTP_INCOMING_COOKIES] = req.cookies
   }
 
+  // TODO: no need to analyze it if it was already done by the body-parser hook
+  if (req.query !== undefined && req.query !== null) {
+    payload[addresses.HTTP_INCOMING_QUERY] = req.query
+  }
+
   waf.run(payload, req)
 
   waf.disposeContext(req)
 
   Reporter.finishRequest(req, res)
 }
 
-function onRequestBodyParsed ({ req, res, abortController }) {
+function onRequestBodyParsed ({ req, res, body, abortController }) {
+  if (!req) {
+    const store = storage.getStore()
+    req = store?.req
```

Comment:
Should we obtain the `body` from the `req` obtained from the `store` at this point?

Reply:
body is not necessarily in the request object.
In next, it is not for some configurations. This is the reason why are we sending body as parameter in the channel.
The subscriber don't need to know about the details of the framework.

Reply:
ah, then wouldn't be better to check `if (body === undefined || body === null) return` at the beginning of the method?